src/main/java/org/openstreetmap/atlas/checks/flag/serializer/GeoJsonFeatureToAtlasEntityConverter.java

```diff
@@ -0,0 +1,203 @@
+package org.openstreetmap.atlas.checks.flag.serializer;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeSet;
+
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.builder.RelationBean;
+import org.openstreetmap.atlas.geography.atlas.complete.CompleteArea;
+import org.openstreetmap.atlas.geography.atlas.complete.CompleteEdge;
+import org.openstreetmap.atlas.geography.atlas.complete.CompleteLine;
+import org.openstreetmap.atlas.geography.atlas.complete.CompleteNode;
+import org.openstreetmap.atlas.geography.atlas.complete.CompletePoint;
+import org.openstreetmap.atlas.geography.atlas.complete.CompleteRelation;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.geojson.GeoJsonUtils;
+import org.openstreetmap.atlas.geography.geojson.parser.domain.feature.Feature;
+import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.LineString;
+import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.MultiPolygon;
+import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.Point;
+
+/**
+ * A GeoJson {@link Feature} to an {@link AtlasEntity} converter.
+ *
+ * @author bbreithaupt
+ */
+public final class GeoJsonFeatureToAtlasEntityConverter
+{
+    private static final List<String> metaProperties = Arrays.asList(GeoJsonUtils.ITEM_TYPE,
+            GeoJsonUtils.IDENTIFIER, GeoJsonUtils.OSM_IDENTIFIER, "relations", "members", "flag:id",
+            "flag:type");
+    private static final Long DEFAULT_IDENTIFIER = 0L;
+
+    /**
+     * Attempt to convert a GeoJson {@link Feature} to an {@link AtlasEntity}.
+     *
+     * @param feature
+     *            {@link Feature}
+     * @return an {@link Optional} {@link AtlasEntity}
+     */
+    public static Optional<AtlasEntity> convert(final Feature feature)
```

Comment:
Just a thought-- since the flagged object conversion is lossy in terms of relation identifiers and other feature-dependent data, we might want to (if not already) have some way of recovering that or at least merging it with the existing feature when we're applying fix suggestions in MR

Reply:
Initially this should not be a problem while we are focusing on tag change suggestions, but this will likely be something we need to look at for other types of suggestions.